great_expectations/core/batch.py

```diff
@@ -930,7 +934,161 @@ def get_batch_request_as_dict(
+
+    batch_request_as_dict: dict = {
+        "datasource_name": datasource_name,
+        "data_connector_name": data_connector_name,
+        "data_asset_name": data_asset_name,
+        "data_connector_query": data_connector_query,
+        "batch_spec_passthrough": batch_spec_passthrough,
+    }
+
+    deep_filter_properties_iterable(
+        properties=batch_request_as_dict,
+        inplace=True,
+    )
+
+    return BatchRequest(**batch_request_as_dict)
+
+
+def _get_runtime_batch_request(
+    *,
+    datasource_name: Optional[str] = None,
+    data_connector_name: Optional[str] = None,
+    data_asset_name: Optional[str] = None,
+    runtime_parameters: Optional[dict] = None,
+    batch_identifiers: Optional[dict] = None,
+    batch_spec_passthrough: Optional[dict] = None,
+    batch_data: Optional[Any] = None,
+    query: Optional[str] = None,
+    path: Optional[str] = None,
+    **kwargs,
+) -> Optional[RuntimeBatchRequest]:
+    """Returns a `RuntimeBatchRequest`, or `None` if the arguments don't support it."""
+    if any(
+        [batch_data, query, path, runtime_parameters]
+    ):  # one of these must be specified for runtime batch requests
+        # parameter checking
+        if len([arg for arg in [batch_data, query, path] if arg is not None]) > 1:
+            raise ValueError("Must provide only one of batch_data, query, or path.")
+
+        if runtime_parameters and any(
+            [
+                batch_data and "batch_data" in runtime_parameters,
+                query and "query" in runtime_parameters,
+                path and "path" in runtime_parameters,
+            ]
+        ):
+            raise ValueError(
+                "If batch_data, query, or path arguments are provided, the same keys cannot appear in the "
+                "runtime_parameters argument."
+            )
+
+        # build runtime parameters
+        runtime_parameters = runtime_parameters or {}
+        if batch_data:
+            runtime_parameters["batch_data"] = batch_data
+        elif query:
+            runtime_parameters["query"] = query
+        elif path:
+            runtime_parameters["path"] = path
+
+        if batch_identifiers is None:
+            batch_identifiers = kwargs
+        else:
+            # TODO: Raise a warning if kwargs exist
+            pass
+
+        batch_request_as_dict = {
+            "datasource_name": datasource_name,
+            "data_connector_name": data_connector_name,
+            "data_asset_name": data_asset_name,
+            "runtime_parameters": runtime_parameters,
+            "batch_identifiers": batch_identifiers,
+            "batch_spec_passthrough": batch_spec_passthrough,
+        }
+
+        deep_filter_properties_iterable(
+            properties=batch_request_as_dict,
+            inplace=True,
+        )
+
+        return RuntimeBatchRequest(**batch_request_as_dict)
+    else:
+        return None
+
+
+def get_batch_request_from_acceptable_arguments(
```

Comment:
this was a very complex method (complexity 21) because it was intermingling parameters and conditions for runtime, block-style, and (unknowingly) fluent batch requests. I broke out the methods for runtime and block-config batch requests, isolating the relevant checks and parameters to each method respectively. Lots of bugs were found and fixed as part of this simplification

Reply:
@donaldheppner -- very nice!  Thanks!  However, the FDS clause (whoever added it), had to be "knowingly" (otherwise, why would it be here)?

Reply:
that's the problem with complex methods, you can't possibly know what you're doing when you make changes to them